packages/babel-plugin-transform-typescript/src/enum.ts

```diff
@@ -124,7 +125,29 @@ export function translateEnumValues(
           value = t.stringLiteral(constValue);
         }
       } else {
-        value = initializer;
+        const IdentifierVisitor: Visitor = {
+          Identifier(expr) {
+            if (seen.has(expr.node.name)) {
+              expr.replaceWith(
+                t.memberExpression(
+                  t.cloneNode(path.node.id),
+                  t.cloneNode(expr.node),
+                ),
+              );
+              expr.skip();
+            }
+          },
+          MemberExpression(expr) {
+            expr.skip();
+          },
```

Comment:
I think that skipping member expressions breaks transpilation of this code:
```js
enum A {
    a = 0,
    b = 1,
    // @ts-ignore
    x = a.toString(),
}
```

You can try visiting `ReferencedIdentifier` instead of `Identifier`, so that it only visits the "correct" identifiers and you don't need to manually skip member expressions.